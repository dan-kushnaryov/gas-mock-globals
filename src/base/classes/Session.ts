import User from './User';

export default class Session {
  static getActiveUser() {
    return new User();
  }

  static getActiveUserLocale() {
    return 'en';
  }

  static getEffectiveUser() {
    return new User();
  }

  static getScriptTimeZone() {
    return 'GMT-11:00';
  }

  static getTemporaryActiveUserKey() {
    return '5e93753f-994c-4cb7-8ec1-9dc14877da69';
  }
}
