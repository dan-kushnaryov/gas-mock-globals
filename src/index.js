// Global Services
const Utilities = require('./utilities/Utilities')
const ScriptApp = require('./script/ScriptApp')
const OAuth2 = require('./auth/OAuth2')
const CardService = require('./card/CardService')
const CacheService = require('./cache/CacheService')
const PropertiesService = require('./properties/PropertiesService')
const UrlFetchApp = require('./url-fetch/UrlFetchApp')
const UrlFetchAppStubConfiguration = require('./url-fetch/classes/UrlFetchAppStubConfiguration')
const Session = require('./base/classes/Session')
const Logger = require('./base/classes/Logger')

global.Utilities = Utilities
global.ScriptApp = ScriptApp
global.OAuth2 = OAuth2
global.CardService = CardService
global.CacheService = CacheService
global.PropertiesService = PropertiesService
global.UrlFetchApp = UrlFetchApp
global.Session = Session
global.Logger = Logger
global.UrlFetchAppStubConfiguration = UrlFetchAppStubConfiguration