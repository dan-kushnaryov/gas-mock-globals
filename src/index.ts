// Global Services
import OAuth2 from './auth/OAuth2';
import Logger from './base/classes/Logger';
import Session from './base/classes/Session';
import CacheService from './cache/CacheService';
import CardService from './card/CardService';
import PropertiesService from './properties/PropertiesService';
import ScriptAppStubConfiguration from './script/classes/ScriptAppStubConfiguration';
import ScriptApp from './script/ScriptApp';
import SpreadsheetStubConfiguration from './spreadsheet/classes/SpreadsheetStubConfiguration';
import Spreadsheet from './spreadsheet/Spreadsheet';
import SpreadsheetApp from './spreadsheetapp/SpreadsheetApp';
import HttpResponse from './url-fetch/classes/HttpResponse';
import UrlFetchAppStubConfiguration from './url-fetch/classes/UrlFetchAppStubConfiguration';
import UrlFetchApp from './url-fetch/UrlFetchApp';
import Utilities from './utilities/Utilities';

global.Utilities = Utilities;
global.ScriptApp = ScriptApp;
global.OAuth2 = OAuth2;
global.CardService = CardService;
global.CacheService = CacheService;
global.PropertiesService = PropertiesService;
global.UrlFetchApp = UrlFetchApp;
global.HttpResponse = HttpResponse;
global.Session = Session;
global.Logger = Logger;
global.UrlFetchAppStubConfiguration = UrlFetchAppStubConfiguration;
global.SpreadsheetApp = SpreadsheetApp;
global.Spreadsheet = Spreadsheet;
global.SpreadsheetStubConfiguration = SpreadsheetStubConfiguration;
global.ScriptAppStubConfiguration = ScriptAppStubConfiguration;
