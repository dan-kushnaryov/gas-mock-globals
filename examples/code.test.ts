/* eslint-disable no-undef */
import '../src';
import Session from '../src/base/classes/Session';
import { doesUserSpeakEnglish, dummyFunction, getGmailCard } from './code';

test('Function which uses Apps Script services under the hood', () => {
  expect(dummyFunction()).toBe("i'm dummy");
});

test('Mocking global services methods', () => {
  Session.getActiveUserLocale = jest.fn().mockReturnValue('fr');

  const result = doesUserSpeakEnglish();

  expect(Session.getActiveUserLocale).toBeCalled();
  expect(result).toBeFalsy();
});

test('Gmail cards snapshot', () => {
  const card = getGmailCard();
  expect(card.printJson()).toMatchSnapshot();
});
