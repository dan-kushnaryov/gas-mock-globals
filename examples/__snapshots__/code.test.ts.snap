// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gmail cards snapshot 1`] = `
Object {
  "cardActions": Array [],
  "header": Object {
    "title": "Card title",
  },
  "name": "Card name",
  "sections": Array [
    Object {
      "header": "Section header",
      "widgets": Array [
        Object {
          "text": "Text paragraph",
        },
      ],
    },
  ],
}
`;
